snuba/datasets/schemas/__init__.py

```diff
@@ -45,6 +45,18 @@ def get_mandatory_conditions(self) -> Sequence[Condition]:
         """
         raise NotImplementedError
 
+    @abstractmethod
```

Comment:
I moved the pre-where keys as part of the attribute of the data source (whether it is a table, a join or anything else).
This is because pre-where conditions go together with the table we are querying. During query processing we can swap this query, thus the pre where column can change. This way the list is always attached to the table we are querying.

Reply:
Good observation, this makes a lot of sense.